src/bebone/gfx/window/events/mouse_listener.cpp

```diff
@@ -1,11 +1,11 @@
-#include "mouse_listener.h"
-
-namespace bebone::core {
-    MouseListener::MouseListener(std::shared_ptr<Input>& input) : input(input) { }
-
-    void MouseListener::operator()(InputMouseButtonEvent &event) {
-        const auto key_code = static_cast<KeyCode>(event.button);
-        const auto input_type = static_cast<InputType>(event.action);
-        input->queue_key(key_code, input_type);
-    }
+#include "mouse_listener.h"
+
+namespace bebone::gfx {
```

Comment:
I am not sure is it good idea of storing mouse listener in gfx module, just not makes a lot of sense, but for now I guess there is no better solution

![image](https://github.com/Maksasj/bebone/assets/36482401/41d15f4d-88aa-48cf-88ff-5cbacd61e768)

Reply:
I moved it to gfx module, because MouseListener class inherits from EventListener<InputMouseButtonEvent>. InputMouseButtonEvent is a window event.